### PR TITLE
update test_nlp.py and nlp.py to adapt to TF 2.0 

### PR DESCRIPTION
**Pull request by OliverZijia**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by OliverZijia** [comment]
update test_nlp.py and nlp.py to adapt to TF 2.0

File: tensorlayer/nlp.py [hunk 1]
```diff
@@ -784,7 +784,7 @@ def word_ids_to_words(data, id_to_word):
 
     Examples
     ---------
-    see ``tl.nlp.words_to_word_ids``
+    >>> see ``tl.nlp.words_to_word_ids``
```
Comment (JingqingZ) [inline tensorlayer/nlp.py]:
```python
>>> # see ``tl.nlp.words_to_word_ids``
```
There should be # as it is a comment

File: tensorlayer/nlp.py [hunk 2]
```diff
@@ -824,7 +824,7 @@ def save_vocab(count=None, name='vocab.txt'):
     vocabulary_size = len(count)
     with open(os.path.join(pwd, name), "w") as f:
         for i in xrange(vocabulary_size):
-            f.write("%s %d\n" % (tf.compat.as_text(count[i][0]), count[i][1]))
+            f.write("%s %d\n" % (tf.compat.v2.compat.as_text(count[i][0]), count[i][1]))
```
Comment (JingqingZ) [inline tensorlayer/nlp.py]: We may prefer to use latest API in TF2 and not to use any code under `tf.compat`. Is there an equivalent TF2 API for `as_text`?

File: tensorlayer/nlp.py [hunk 3]
```diff
@@ -862,7 +862,7 @@ def basic_tokenizer(sentence, _WORD_SPLIT=re.compile(b"([.,!?\"':;)(])")):
 
     """
     words = []
-    sentence = tf.compat.as_bytes(sentence)
+    sentence = tf.compat.v2.compat.as_bytes(sentence)
```
Comment (JingqingZ) [inline tensorlayer/nlp.py]: Same problem as mentioned above. `tf.compat` should not be used.

File: tensorlayer/nlp.py [hunk 4]
```diff
@@ -969,7 +969,7 @@ def initialize_vocabulary(vocabulary_path):
         rev_vocab = []
         with gfile.GFile(vocabulary_path, mode="rb") as f:
             rev_vocab.extend(f.readlines())
-        rev_vocab = [tf.compat.as_bytes(line.strip()) for line in rev_vocab]
+        rev_vocab = [tf.compat.v2.compat.as_bytes(line.strip()) for line in rev_vocab]
```
Comment (JingqingZ) [inline tensorlayer/nlp.py]: Same problem as mentioned above.

File: tensorlayer/nlp.py (hunk 3 above)
Comment (zsdonghao) [reply tensorlayer/nlp.py]: it is the same with  xxx.join(" ")??

File: tensorlayer/nlp.py (hunk 2 above)
Comment (zsdonghao) [reply tensorlayer/nlp.py]: what is this function for? i believe Python can do it already.

File: tests/test_nlp.py [hunk 5]
```diff
@@ -23,6 +24,36 @@ def setUpClass(cls):
...
+        data, count, dictionary, reverse_dictionary = tl.nlp.build_words_dataset(words, vocabulary_size, True)
+        tl.nlp.save_vocab(count, name='vocab_text8.txt')        
+
+    def test_basic_tokenizer(self):
+        train_path = "wmt/giga-fren.release2"
+        with gfile.GFile(train_path + ".en", mode="rb") as f:
+           for line in f:
+              tokens = tl.nlp.basic_tokenizer(line)
+              tl.logging.info(tokens)
+              exit()
```
Comment (JingqingZ) [inline tests/test_nlp.py]: You may use `break` instead.

**Comment by OliverZijia** [comment]
the content of vocab_text8.txt(part):

UNK 418391
the 1061396
of 593677
and 416629
one 411764
in 372201
a 325873
to 316376
zero 264975
nine 250430
two 192644
is 183153
as 131815
eight 125285
for 118445
s 116710
five 115789
three 114775
was 112807
by 111831
that 109510
four 108182
six 102145
seven 99683
with 95603
on 91250
are 76527
it 73334
from 72871
or 68945
his 62603
an 61925
be 61281
this 58832

**Pull request merged**